src/ddmd/cond.d

```diff
@@ -508,12 +508,20 @@ extern (C++) final class StaticIfCondition : Condition
                 printf("\ts = '%s', kind = %s\n", sds.toChars(), sds.kind());
             }
         }
+
+        int errorReturn()
+        {
+            if (!global.gag)
+                inc = 2; // so we don't see the error message again
+            return 0;
+        }
+
         if (inc == 0)
         {
             if (exp.op == TOKerror || nest > 100)
             {
                 error(loc, (nest > 1000) ? "unresolvable circular static if expression" : "error evaluating static if expression");
-                goto Lerror;
+                return errorReturn();
```

Comment:
I called it `errorReturn()` because such is already used elsewhere for that purpose, and because `error` is already heavily overloaded and I didn't want to add confusion.